src/components/Table.js

```diff
@@ -224,15 +224,24 @@ export default class Table extends React.Component<TableDefaultProps, TableProps
 
     // If the parent provided an update hook, initialize the parent with selection values.
     if (onSelect) {
-      onSelect([], activeRowIndex);
+      const activeRowIndex = sortedRows.length > 0 ? 0 : null;
+      const selectedRow = sortedRows.length > 0 ? sortedRows[0] : null;
+      const selectedRows = selectedRow ? [selectedRow] : [];
+
+      onSelect(selectedRows, selectedRow);
+      // TODO: Look at redux/reselect pattern as alternative to manipulating data coming from api for component consumption.
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({
+        activeRowIndex,
+        anchorRowIndex: activeRowIndex,
+        selectedRowIndices: sortedRows.length > 0 ? new Set(selectedRows) : new Set(),
+      });
     }
   }
 
   componentWillReceiveProps(newProps: TableProps) {
     // If a rowComparator is specified, use it, otherwise, do a deep row comparison and reset selection if anything changed.
-    if ((newProps.rows && this.props.rows && newProps.rows.length !== this.props.rows.length)
-      || (newProps.rows && !this.props.rows)
-      || (!newProps.rows && this.props.rows)
+    if (newProps.rows.length !== this.props.rows.length
```

Comment:
I have a question about the way you're dealing with selections in this method: If a row "Foo" was previously selected and it was in the third position in the table (e.g., sortedRows[2]), and then the order of the rows changes so "Foo in at the 5th position (sortedRows[4]), won't it no longer be selected (and whatever is now at sortedRows[2] will be)? This would happen if the parent sorts the rows according to a different column, say... As a user, I think I would expect Foo to remain selected as I change the sort order (as long as it's still in the table).

Reply:
When we discussed this with Dorit, it was my understanding that we decided that selection should be reset when sorting. filtering, search, or paging happens, if that's ok with you. I'd prefer to get this in for now. If you feel strongly about it we can raise it in the meeting today.